libs/components/src/lib/split-button/split-button.spec.ts

```diff
@@ -0,0 +1,140 @@
+			expect(element.size).toBeUndefined();
+			expect(element.indicator).toBeInstanceOf(HTMLButtonElement);
+			expect(element.action).toBeInstanceOf(HTMLButtonElement);
+		});
+	});
+
+	describe('icon', () => {
+		it('adds an icon to the button', async () => {
+			element.icon = 'home';
+			await elementUpdated(element);
+
+			const icon = element.shadowRoot?.querySelector(ICON_SELECTOR) as Icon;
+			expect(icon).toBeInstanceOf(Icon);
+			expect(icon.name).toEqual('home');
+		});
+	});
+
+	describe('split indicator', () => {
+		it('adds a split indicator to the button', async () => {
+			element.splitIndicator = 'home';
+			await elementUpdated(element);
+
+			const icon = element.indicator.querySelector(ICON_SELECTOR) as Icon;
+			expect(icon).toBeInstanceOf(Icon);
+			expect(icon.name).toEqual('home');
+		});
+	});
+
+	describe('label', () => {
+		it('set label property to node', async () => {
+			const label = 'lorem';
+			element.label = label;
+			await elementUpdated(element);
+
+			const control = element.shadowRoot?.querySelector('.control');
+			expect(control?.textContent?.trim()).toEqual(label);
+		});
+	});
+
+	describe('connotation', () => {
+		it('sets correct internal connotation style', async () => {
+			const connotation = 'cta';
+			(element as any).connotation = connotation;
+			await elementUpdated(element);
+
+			const control = element.shadowRoot?.querySelector(`.control.connotation-${connotation}`);
+			expect(control).toBeInstanceOf(Element);
+		});
+	});
+
+	describe('shape', () => {
+		it('sets correct internal shape style', async () => {
+			const shape = 'pill';
+			(element as any).shape = shape;
+			await elementUpdated(element);
+
+			const control = element.shadowRoot?.querySelector(`.control.shape-${shape}`);
+			expect(control).toBeInstanceOf(Element);
+		});
+	});
+
+	describe('appearance', () => {
+		it('sets correct internal appearance style', async () => {
+			const appearance = 'filled';
+			(element as any).appearance = appearance;
+			await elementUpdated(element);
+
+			const control = element.shadowRoot?.querySelector(`.control.appearance-${appearance}`);
+			expect(control).toBeInstanceOf(Element);
+		});
+	});
+
+	describe('size', () => {
+		it('sets correct internal size style', async () => {
+			const size = 'condensed';
+			(element as any).size = size;
+			await elementUpdated(element);
+
+			const control = element.shadowRoot?.querySelector(`.control.size-${size}`);
+			expect(control?.classList.contains(`size-${size}`)).toBeTruthy();
+		});
+	});
+
+	describe('disabled', function () {
+		it('should set disabled class when disabled is true', async () => {
+			const appearance = 'filled';
+			(element as any).appearance = appearance;
+			element.toggleAttribute('disabled', true);
+			await elementUpdated(element);
+
+			const control = element.shadowRoot?.querySelector(`.control.appearance-${appearance}.disabled`);
+			expect(control).toBeInstanceOf(Element);
```

Comment:
I think this is more explicit:
```suggestion
			const controlHasDisabledClass = getControlElement(element).classList.contains(`appearance-${appearance}.disabled`);
			expect(controlHasDisabledClass).toBeTruthy();
```